nipype/utils/nipype2boutiques.py

```diff
@@ -55,53 +57,133 @@ def generate_boutiques_descriptor(
-        tool_desc['inputs'].append(input)
-        tool_desc['command-line'] += input['command-line-key'] + " "
-        if verbose:
-            print("-> Adding input " + input['name'])
+        inp = get_boutiques_input(inputs, interface, name, spec, verbose, ignore_inputs=ignore_inputs)
+        # Handle compound inputs (inputs that can be of multiple types and are mutually exclusive)
+        if inp is None:
+            continue
+        if isinstance(inp, list):
+            mutex_group_members = []
+            tool_desc['command-line'] += inp[0]['value-key'] + " "
+            for i in inp:
+                tool_desc['inputs'].append(i)
+                mutex_group_members.append(i['id'])
+                if verbose:
+                    print("-> Adding input " + i['name'])
+            # Put inputs into a mutually exclusive group
+            tool_desc['groups'].append({'id': inp[0]['id'] + "_group",
+                                        'name': inp[0]['name'] + " group",
+                                        'members': mutex_group_members,
+                                        'mutually-exclusive': True})
+        else:
+            tool_desc['inputs'].append(inp)
+            tool_desc['command-line'] += inp['value-key'] + " "
+            if verbose:
+                print("-> Adding input " + inp['name'])
+
+    # Generates input groups
+    tool_desc['groups'] += get_boutiques_groups(interface.inputs.traits(transient=None).items())
+    if len(tool_desc['groups']) == 0:
+        del tool_desc['groups']
 
     # Generates tool outputs
+    generate_tool_outputs(outputs, interface, tool_desc, verbose, True)
+
+    # Generate outputs with various different inputs to try to generate
+    # as many output values as possible
+    custom_inputs = generate_custom_inputs(tool_desc['inputs'])
+
+    for input_dict in custom_inputs:
+        interface = getattr(module, interface_name)(**input_dict)
+        outputs = interface.output_spec()
+        generate_tool_outputs(outputs, interface, tool_desc, verbose, False)
+
+    # Fill in all missing output paths
+    for output in tool_desc['output-files']:
+        if output['path-template'] == "":
+            fill_in_missing_output_path(output, output['name'], tool_desc['inputs'])
+
+    # Add tags
+    desc_tags = {
+        'domain': 'neuroinformatics',
+        'source': 'nipype-interface'
+    }
+
+    if tags is not None:
+        tags_dict = json.loads(tags)
+        for k, v in tags_dict.items():
+            if k in desc_tags:
+                if not isinstance(desc_tags[k], list):
+                    desc_tags[k] = [desc_tags[k]]
+                desc_tags[k].append(v)
+            else:
+                desc_tags[k] = v
+
+    tool_desc['tags'] = desc_tags
+
+    # Check for positional arguments and reorder command line args if necessary
+    tool_desc['command-line'] = reorder_cmd_line_args(tool_desc['command-line'], interface, ignore_inputs)
```

Comment:
rf: line length

Reply:
@mgxd What should be the max line length?

Reply:
We should follow [pep8](https://www.python.org/dev/peps/pep-0008/) conventions, so 79 characters